homeassistant/components/mopar/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for the Mopar vehicle sensor platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mopar/
+"""
+
+from homeassistant.components.mopar import DOMAIN as MOPAR_DOMAIN, DATA_UPDATED
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, LENGTH_KILOMETERS)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['mopar']
+
+ATTR_VEHICLE_INDEX = 'vehicle_index'
+ICON = 'mdi:car'
+
+
+async def async_setup_platform(hass, config, add_entities,
+                               discovery_info=None):
+    """Set up the Mopar platform."""
+    data = hass.data[MOPAR_DOMAIN]
+    add_entities([MoparSensor(data, index)
+                  for index, _ in enumerate(data.vehicles)], True)
+
+
+class MoparSensor(Entity):
+    """Mopar vehicle sensor."""
+
+    def __init__(self, data, index):
+        """Initialize the sensor."""
+        self._index = index
+        self._vehicle = {}
+        self._vhr = {}
+        self._tow_guide = {}
+        self._odometer = None
+        self._data = data
+        self._name = self._data.get_vehicle_name(self._index)
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        async_dispatcher_connect(
+            self.hass, DATA_UPDATED, self._schedule_immediate_update
+        )
+
+    def update(self):
+        """Update device state."""
+        self._data.update()
```

Comment:
How is the update flow supposed to work? Looks like it's cyclic now but also throttled. It doesn't look like this will continue to work after it hits the first throttle.

Reply:
What I was trying to do is have the component cache the data in the MoparData object. Each platform would call update (first one to hit the throttle would actually force the cache to update), each subsequent platform would read from the cache. I think I need to remove the `dispatcher_send` call to make this work correctly though right?

Reply:
We have to decide if we want the platforms to poll at an interval, or if the component should do that. At the moment the platforms have `should_poll` to `False` and there's no custom polling started from the component. Either of those have to change.

Reply:
Yep, I totally forgot I was intending to have the component poll and passing that via signal to the sensor platform. Fixed, thanks for the reminder!